src/idpyoidc/server/session/grant.py

```diff
@@ -235,9 +233,11 @@ def payload_arguments(
                 secondary_identifier=secondary_identifier,
             )
 
-        user_id, _, _ = endpoint_context.session_manager.decrypt_session_id(session_id)
-        user_info = endpoint_context.claims_interface.get_user_claims(user_id, _claims_restriction)
-        payload.update(user_info)
+        if endpoint_context.session_manager.node_type[0] == "user":
```

Comment:
do we have a check for a void node_type?
[0] would fail

is this an critical exception or it is handled on top, in the calling code?

Reply:
That should be handled when the Session Manager is initiated.
You could possibly break this with a configuration error.
I'll add a check.